concrete/src/Job/Factory.php

```diff
@@ -0,0 +1,240 @@
+<?php
+namespace Concrete\Core\Job;
+
+use Concrete\Core\Application\Application;
+use Concrete\Core\Package\PackageList;
+
+class Factory
+{
+    /** @var \Concrete\Core\Application\Application */
+    private $app;
+
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * @param bool $scheduledOnly
+     *
+     * @return Job[]
+     */
+    public function getList($scheduledOnly = false)
+    {
+        $db = $this->app['database']->connection();
+
+        if ($scheduledOnly) {
+            $q = "SELECT jID FROM Jobs WHERE isScheduled = 1 ORDER BY jDateLastRun, jID";
+        } else {
+            $q = "SELECT jID FROM Jobs ORDER BY jDateLastRun, jID";
+        }
+
+        $r = $db->query($q);
+        $jobs = [];
+        while ($row = $r->fetchRow($q)) {
+            $j = $this->getByID($row['jID']);
+            if (is_object($j)) {
+                $jobs[] = $j;
+            }
+        }
+
+        return $jobs;
+    }
+
+    /**
+     * @param int $jID
+     *
+     * @return null|Job
+     */
+    public function getByID($jID = 0)
+    {
+        $db = $this->app['database']->connection();
+        $jobData = $db->fetchAssoc("SELECT * FROM Jobs WHERE jID=?", [
+            intval($jID),
+        ]);
+
+        if (!$jobData || !$jobData['jHandle']) {
+            return null;
+        }
+
+        return $this->getJobObjByHandle($jobData['jHandle'], $jobData);
+    }
+
+    /**
+     * @param string $jHandle
+     *
+     * @return null|Job
+     */
+    public function getByHandle($jHandle = '')
+    {
+        $db = $this->app['database']->connection();
+        $jobData = $db->fetchAssoc("SELECT * FROM Jobs WHERE jHandle=?", [
+            $jHandle,
+        ]);
+
+        if (!$jobData || !$jobData['jHandle']) {
+            return null;
+        }
+
+        return $this->getJobObjByHandle($jobData['jHandle'], $jobData);
+    }
+
+    /**
+     * @param string $jHandle
+     * @param array $jobData
+     *
+     * @return null|Job
+     */
+    public function getJobObjByHandle($jHandle = '', $jobData = [])
```

Comment:
I don't think this is worth having on this class. We can recreate this functionality in the old method like this:

```php
$job = $factory->getByHandle($handle);
if ($job && $jobData) {
    $job->setPropertiesFromArray($jobData);
}
return $job;
```